backend/lib/services/terminals/index.js

```diff
@@ -237,53 +230,93 @@ async function getTargetCluster ({ user, namespace, name, target }) {
-      const seedName = shootResource.spec.cloud.seed
-      const seed = _.find(getSeeds(), ['metadata.name', seedName])
+  hostCluster.namespace = seedShootNS
+  hostCluster.secretRef = _.get(seed, 'spec.secretRef')
+  hostCluster.kubeApiServer = await getKubeApiServerHostForSeed({ gardenClient, coreClient: gardenCoreClient, shootsService: shoots, seed })
+  return hostCluster
+}
 
-      hostCluster.namespace = seedShootNS
-      hostCluster.secretRef = _.get(seed, 'spec.secretRef')
-      hostCluster.kubeApiServer = await getKubeApiServerHostForSeed({ user, seed })
-    } else { // as enduser - host cluster is the shoot
-      assert.strictEqual(target, TARGET.shoot, 'unexpected target')
+async function getShootHostCluster ({ gardenClient, gardenCoreClient, namespace, name, target, body }) {
+  assert.strictEqual(target, TargetEnum.SHOOT, 'unexpected target')
 
-      hostCluster.namespace = undefined // this will create a temporary namespace
-      hostCluster.secretRef = {
-        namespace,
-        name: `${name}.kubeconfig`
-      }
-      hostCluster.kubeApiServer = await getKubeApiServerHostForShoot({ user, shootResource })
-    }
+  const hostCluster = {}
+  hostCluster.config = getConfigFromBody(body)
+
+  const shootResource = await shoots.read({ gardenClient, namespace, name })
+  hostCluster.isHostOrTargetHibernated = _.get(shootResource, 'spec.hibernation.enabled', false)
+
+  hostCluster.namespace = undefined // this will create a temporary namespace
+  hostCluster.secretRef = {
+    namespace,
+    name: `${name}.kubeconfig`
   }
+  hostCluster.kubeApiServer = await getKubeApiServerHostForShoot({ gardenClient, coreClient: gardenCoreClient, shootResource })
   return hostCluster
 }
 
-async function createTerminal ({ isAdmin, dashboardClient, user, namespace, name, target, hostCluster, targetCluster }) {
-  const containerImage = getContainerImage(isAdmin)
+function getImageConfigFromBody (body) {
+  return _.pick(body, ['containerImage'])
+}
+
+function getConfigFromBody (body) {
+  return _.pick(body, ['node', 'containerImage', 'privileged', 'hostPID', 'hostNetwork'])
+}
+
+function getHostCluster ({ user, namespace, name, target, body }) {
+  const gardenClient = Garden(user)
+  const gardenCoreClient = Core(user)
+
+  if (target === TargetEnum.GARDEN) {
+    return getGardenTerminalHostCluster({ gardenClient, gardenCoreClient, body })
+  }
+
+  const defaultHost = user.isAdmin ? 'seed' : 'shoot'
+  const preferredHost = _.get(body, 'preferredHost', defaultHost)
+  if (user.isAdmin && preferredHost === 'seed') { // admin only - host cluser is the seed
+    return getSeedHostCluster({ gardenClient, gardenCoreClient, namespace, name, target, body })
+  }
+
+  // host cluster is the shoot
+  return getShootHostCluster({ gardenClient, gardenCoreClient, namespace, name, target, body })
+}
+
+async function createTerminal ({ dashboardClient, user, namespace, name, target, hostCluster, targetCluster }) {
```

Comment:
remove `dashboard.gardener.cloud/targetNamespace` annotation as it is not used anywhere

Reply:
done